src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -1010,6 +1012,31 @@ public static Text of(Object... objects) {
     }
 
     /**
+     * Gets an optional {@link Text} from the provided
+     * {@link TranslationDictionary} by key.
+     *
+     * @param dictionary The translation dictionary to retrieve from
+     * @param key The translation key
+     * @return The text, if present, {@link Optional#empty()} otherwise
+     */
+    public static Optional<Text> get(TranslationDictionary dictionary, String key) {
```

Comment:
Dictionaries should be available to be used as services so you can just pass a plugin instance here.